src/components/TradeWidget/index.tsx

```diff
@@ -425,6 +425,26 @@ const preprocessTokenAddressesToAdd = (addresses: (string | undefined)[], networ
   return tokenAddresses
 }
 
+function buildUrl(params: {
+  sell: string
+  price: string
+  from: string
+  expires: string
+  sellSymbol: string
+  buySymbol: string
+}): string {
+  const { sell, price, from, expires, sellSymbol, buySymbol } = params
+
+  const searchQuery = buildSearchQuery({
+    sell,
+    price,
+    from,
+    expires,
+  })
+
+  return `/trade/${encodeURIComponent(sellSymbol)}-${encodeURIComponent(buySymbol)}?${searchQuery}`
```

Comment:
Here's where you'll do `${encodeURIComponent(symbol).replace(dash, ...)}`, not the whole url

Reply:
ohhhhh got it!

yep, agree :)

Reply:
What if the browser disagrees with encoding `-` and replaces `%2D` with a dash anyway in the URL? :/

Reply:
What do you mean disagrees? Browser doesn't do that. What you put in is what gets displayed. Browser doesn't care for anything except url specific symbols like ?#/ that delineate path/?search#hash and locale specific letters ( urls in languages other than English, which is a horrible idea IMHO)

Reply:
% in url is just a %, to decode or take at face value is up to us not browser. Even react-router doesn't automatically decode url params, though it's a frequent subject of issues in their repo

Reply:
Well, something is changing `%2D` back to a `-`.
Try it out now on localhost.

I left a console.log to print the generated URL.

That has the escaped version, but the url still shows a `-`.
![screenshot_2020-04-16_14-15-51](https://user-images.githubusercontent.com/43217/79507557-c9a0f600-7fec-11ea-971b-e49045012bbd.png)
![screenshot_2020-04-16_14-16-00](https://user-images.githubusercontent.com/43217/79507561-cc035000-7fec-11ea-8480-6b36f42f75e0.png)

Reply:
Shit, you are right, browser auto-decodes `-~'` symbols, so we can't rely on replacing them
Then we'll have to change what we use in `path="/trade/:sell-:buy"`, can choose from `$&`:<>[]{}"+%@/;=\^|\',` symbols
```sh
> encodeURIComponent(' $&`:<>[]{}"+#%@/;=?\^|,~\'-')
"%20%24%26%60%3A%3C%3E%5B%5D%7B%7D%22%2B%23%25%40%2F%3B%3D%3F%5E%7C%2C~'-"
```
As you can see `-~'` stayed unencoded

on the bright side, won't have to do `replace(dash`

Reply:
@anxolin @alfetopito @W3stside 
need to decide if we are changing url scheme because some tokens may include dash `-` and browser auto-decodes it from under our feet

Reply:
See #936 for a working example

Reply:
I think this is an edge case, I wouldn't change our beautiful scheme just for that.
If there's not a better solution, I even prefer to do the decoding in two steps using, using `%_2D` instead of `%2D` to encode the parameters
But proably there's a more elegant solution....

Reply:
As I mentioned on https://github.com/gnosis/dex-react/pull/936#issuecomment-615295388.
I'm in favor of using `|` instead, keeping keeping a redirect rule for `-` for backward compatibility.

Although, Anxo's proposal might work...
`replace('-', '%dashdonotreplaceyoudamnpokyingbrowser')`

Reply:
Much better, right?

![screenshot_2020-04-17_10-29-23](https://user-images.githubusercontent.com/43217/79597139-9a46c380-8096-11ea-9d59-bdb831bc9e58.png)